cpp/src/groupby/groupby.cu

```diff
@@ -211,5 +212,25 @@ detail::sort::sort_groupby_helper& groupby::helper()
   return *_helper;
 };
 
+std::pair<std::unique_ptr<table>, std::unique_ptr<column>> groupby::shift(
+  column_view const& values,
+  size_type offset,
+  scalar const& fill_value,
+  rmm::mr::device_memory_resource* mr)
+{
+  CUDF_FUNC_RANGE();
+  CUDF_EXPECTS(values.type() == fill_value.type(),
+               "values and fill_value should have the same type.");
+
+  auto stream         = rmm::cuda_stream_default;
+  auto sorted_keys    = helper().sorted_keys(stream, mr);
```

Comment:
I think you can inline this and rely on RVO rather than moving the output, right?